target_postgres/connector.py

```diff
@@ -174,3 +176,76 @@ def get_column_add_ddl(
                 "column_type": column.type.compile(dialect=self._engine.dialect),
             },
         )
+
+    def get_sqlalchemy_query(self, config: dict) -> dict:
+        """Get query values to be used for sqlalchemy URL creation.
+
+        Args:
+            config: The configuration for the connector.
+
+        Returns:
+            A dictionary with key-value pairs for the sqlalchemy query.
+        """
+        query = {}
+
+        # ssl_enable is for verifying the server's identity to the client.
+        if config["ssl_enable"]:
+            ssl_mode = config["ssl_mode"]
+            query.update({"sslmode": ssl_mode})
+
+            query.update(
+                self.filepath_or_certificate(
+                    "sslrootcert",
+                    config["ssl_certificate_authority"],
+                    ".secrets/root.crt",
+                )
+            )
+
+        # ssl_client_certificate_enable is for verifying the client's identity to the server.
+        if config["ssl_client_certificate_enable"]:
+            query.update(
+                self.filepath_or_certificate(
+                    "sslcert", config["ssl_certificate"], ".secrets/cert.crt"
+                )
+            )
+            query.update(
+                self.filepath_or_certificate(
+                    "sslkey", config["ssl_private_key"], ".secrets/pkey.key", True
+                )
+            )
+        return query
+
+    def filepath_or_certificate(
+        self,
+        key: str,
+        value: str,
+        alternative_name: str,
+        restrict_permissions: bool = False,
+    ) -> dict[str, str]:
+        """Provide the appropriate key-value pair based on a filepath or raw value.
+
+        For SSL configuration options, support is provided for either raw values in
+        .env file or filepaths to a file containing a certificate. This function
+        attempts to parse a value as a filepath, and if no file is found, assumes the
+        value is a certificate and creates a file named `alternative_name` to store the
+        file.
+
+        Args:
+            key: The key that should be returned in a dictionary.
+            value: Either a filepath or a raw value to be written to a file.
+            alternative_name: The filename to use in case `value` is not a filepath.
+            restrict_permissions: Whether to restrict permissions on a newly created
+                file. On UNIX systems, private keys cannot have public access.
+
+        Returns:
+            A dictionary with key-value pairs for the sqlalchemy query
+
+        """
+        if path.isfile(value):
+            return {key: value}
+        else:
+            with open(alternative_name, "wb") as alternative_file:
+                alternative_file.write(value.encode("utf-8"))
+            if restrict_permissions:
+                chmod(alternative_name, 0o600)
```

Comment:
```suggestion
                chmod(alternative_name, 0x600)
```

Does this work?  Just looks odd to be 0o